apps/meteor/server/startup/migrations/v303.ts

```diff
@@ -0,0 +1,20 @@
+import { Users } from '@rocket.chat/models';
+
+import { settings } from '../../../app/settings/server';
+import { parseCSV } from '../../../lib/utils/parseCSV';
+import { addMigration } from '../../lib/migrations';
+
+addMigration({
+	version: 303,
+	name: 'Add default roles to users without roles field',
+	async up() {
+		const defaultRoles = settings.get('Accounts_Registration_Users_Default_Roles');
+
+		await Users.updateMany(
+			{
+				roles: { $exists: false },
+			},
+			{ $set: { roles: defaultRoles ? parseCSV(String(defaultRoles)) : [] } },
+		);
+	},
```

Comment:
Instead of loading the setting value directly, use `getNewUserRoles` to handle the parsing for you.
If the list is empty, set them to `['user']` instead of an empty array.